_internal/types.ts

```diff
@@ -208,34 +208,18 @@ export type ProviderConfiguration = {
  * ```
  */
 export interface SWRHook {
-  <Data = any, Error = any, SWRKey extends StrictKey = StrictKey>(
+  <Data = any, Error = any, SWRKey extends Key = StrictKey>(
```

Comment:
I use `StrictKey` here because i want 
```ts
useSWR<string, any>('/api/', (key: string) => key)
```  
to match a simpler overload signature

<img width="947" alt="Screenshot 2022-12-15 at 18 06 55" src="https://user-images.githubusercontent.com/15878786/207833173-6ad67dd0-b5a8-4b3b-87f3-3afde4a7ae57.png">

instead of